programming/Encoders.md

```diff
@@ -0,0 +1,53 @@
+given data into another format.
+
+**Of course in Robotics, the encoders we use measure how fast a motor spins.**<br> Which is immeasurably important, it's the reason why our robot can actually turn and still drive straight.  This is because of the fact we can precisely measure how quickly our motor is turning and what angle it's turned in which allows us to make sure we have all our wheels facing the right direction.  Without encoders our robot would easily just end up flipping over due to the fact half of the motors were facing the wrong direction. 
+
+There's lots of different examples of encoders at work all around us, 
+such as:
+  * Elevators - Encoders in elevators tell the elevator's controller device its position so that it can properly align itself with the floor and open up the doors at the correct position.
+  * Cranes - Some large cranes use encoders in order to collect accurate location information to know where it is.
+
+**TL;DR:  Encoders are the reason we can tell our robot to drive at angles.**
+
+**How do Encoders (generally) work?**
+
+<img src="http://encoder.com/core/files/encoder/uploads/images/Encoder-exploded-COLOR-v2.jpg"></img>
+
+There are a lot of encoders in the world at large.  Hardware encoders are made out of a variety of different technologies and tend to vary quite a bit as do their purposes.  The encoder in the example shown above is used to detect the speed of which the motor is spinning.  This specific encoder, taken from <a href="http://encoder.com/blog/company-news/what-is-an-encoder/">encoder.com</a>, has a stream of light which comes from the LED and goes through the "Code Disk".  As the light goes through the "Code Disk" it becomes interrupted as the blades from the "Code Disk" block it from the "Photodetector Assembly" which detects light.  The result is a binary output of either the light being on or the light being off.  This simple value is then passed on to another device, generally called a "controller" which actually interprets the data.  Because the controller can obtain these simple values it can measure how frequently the light is being blocked from which it can determine the speed of the motor.  It can check whether or not the motor is actually running based on if there is no change in the on or off value over a long period of time.
+
+Most optical sensors work like the one pictured above by relying on the interruption of light picked up by another sensor.
+
+## How we use encoders!
+
+We use the *robotpy library* for interfacing to our robot's components with Python, and the library has a class interface to <a href="http://robotpy.readthedocs.io/projects/wpilib/en/latest/wpilib/Encoder.html">the Encoder library</a> (yes they call it the encoder library), and to use it, you first create an instance of Encoder oriented to your hardware encoder, such as: **[EXAMPLES OF ENCODERS IN THE ROBOTPY LIBRARY TO BE ADDED]**
```

Comment:
Since I think you might have another update to this change, shall we update some of the formatting to be more Markdown-like and less HTML? For instance, for this link, how about:

    class interface to [the Encoder library](http://robotpy.readthedocs.io/projects/wpilib/en/latest/wpilib/Encoder.html), (yes they call it...

Or better yet, we can put all the links somewhere at the bottom of the page, so we can read it easier:

    We use the *robotpy library* for interfacing to our robot's components with Python, and the library has a class interface to [the Encoder library][5] (yes they call it the encoder library), and to use it, you first create an instance of Encoder oriented to your hardware encoder, such as:

      [5]: http://robotpy.readthedocs.io/projects/wpilib/en/latest/wpilib/Encoder.html